File: 2015/05/02.py
with open("input.txt", "r") as file: lines = file.read().strip().splitlines()

total = sum(
    1 for s in lines
    if any(s[i:i+2] in s[i+2:] for i in range(len(s) - 1))
    and any(s[i] == s[i+2] for i in range(len(s) - 2))
)

print("Part 02:", total)

File: 2025/01/02.lua
local file = io.open("input.txt", "r")
local rotations, dial, count = {}, 50, 0

for line in file:lines() do
    table.insert(rotations, line:sub(1,1) == "L" and -tonumber(line:sub(2)) or tonumber(line:sub(2)))
end

for _, rot in ipairs(rotations) do
    local n_dial = dial + rot
    count = count 
        + (dial > 0 and n_dial <= 0 and 1 or 0)
        + (dial < 0 and n_dial >= 0 and 1 or 0)
        + math.abs(n_dial // 100)
    dial = n_dial % 100
end

print("Part 02 : " .. count)